core/Piwik.php

```diff
@@ -563,7 +563,14 @@ public static function redirectToModule($newModule, $newAction = '', $parameters
      */
     public static function isValidEmailString($emailAddress)
     {
-        return (preg_match('/^[a-zA-Z0-9_.+-]+@[a-zA-Z0-9_.-]+\.[a-zA-Z]{2,}$/D', $emailAddress) > 0);
+        static $zendEmailValidator;
+        if(!$zendEmailValidator) {
+            $zendEmailValidator = new \Zend_Validate_EmailAddress(array(
+                'mx' => false,
+                'deep' => false
+            ));
+        }
```

Comment:
@diosmosis I'm worried, that if I put `Zend_Validate_EmailAddress` in DI, then another core/plugins developers will call email validation using `Zend_Validate_EmailAddress` from DI, instead of using `Piwik::isValidEmailString()`, and in this way Piwik will depends on Zend/Validate library. I gues this case is not about DI pattern, it's about https://en.wikipedia.org/wiki/Flyweight_pattern

But if this is a common practice for Piwik to use DI container as Flyweighter, then I can do some refactoring. Benaka, are you 100% sure about it?